clang/lib/AST/ExprConstant.cpp

```diff
@@ -7441,6 +7500,28 @@ static bool handleLValueToRValueBitCast(EvalInfo &Info, APValue &DestValue,
   return true;
 }
 
+static bool handleLValueToRValueBitCast(EvalInfo &Info, APValue &DestValue,
+                                        APValue &SourceValue,
+                                        const CastExpr *BCE) {
+  assert(CHAR_BIT == 8 && Info.Ctx.getTargetInfo().getCharWidth() == 8 &&
+         "no host or target supports non 8-bit chars");
+  assert(SourceValue.isLValue() &&
+         "LValueToRValueBitcast requires an lvalue operand!");
+
+  if (!checkBitCastConstexprEligibility(&Info, Info.Ctx, BCE))
+    return false;
```

Comment:
Can we avoid doing this twice (once here and again in `handleRValueToRValueBitCast`? It does a recursive walk of the source and destination types so isn't all that cheap.

Reply:
I did it this way so that `handleLValueToRValueBitCast` would be able to break out early to avoid copying the source LValue into an RValue if the expression wasn't eligible, but now that you mention it it doesn't make much sense to do an expensive check twice in order to optimize the exceptional path. Will remove this from `handleLValueToRValueBitCast`.